crates/bevy_hierarchy/src/one_to_one/event.rs

```diff
@@ -0,0 +1,123 @@
+use core::marker::PhantomData;
+
+use bevy_ecs::{entity::Entity, event::Event};
+
+/// A One-to-One [relationship](crate::OneToOne) [`Event`].
+#[derive(Event)]
+pub enum OneToOneEvent<R> {
+    /// A [relationship](crate::OneToOne) was added between two [entities](Entity).
+    Added(OneToOneEventDetails<R>),
+    /// A [relationship](crate::OneToOne) was removed from two [entities](Entity).
+    Removed(OneToOneEventDetails<R>),
+}
+
+impl<R> OneToOneEvent<R> {
+    /// Create a new [`Added`](OneToOneEvent::Added) [`Event`]
+    pub const fn added(primary: Entity, secondary: Entity) -> Self {
+        Self::Added(OneToOneEventDetails::new(primary, secondary))
+    }
+
+    /// Create a new [`Removed`](OneToOneEvent::Removed) [`Event`]
+    pub const fn removed(primary: Entity, secondary: Entity) -> Self {
+        Self::Removed(OneToOneEventDetails::new(primary, secondary))
+    }
+
+    /// Get the primary [`Entity`] in this [`Event`].
+    /// The primary is the _cause_ of the event, while the secondary is the relation.
+    pub const fn primary(&self) -> Entity {
```

Comment:
Could consider `source` and `target`, even though the edge isn't directed. I'm not terribly fond of primary and secondary, although this is of course bikeshedding.

Reply:
I've decided to adopt the terminology `primary` and `foreign`, as that maps well to the relational database terminology that these changes are most analogous to.

Reply:
Thanks, I hate it! 😉